build-logic/src/main/kotlin/org/terasology/gradology/module_build.kt

```diff
@@ -72,7 +73,15 @@ class ModuleMetadataForGradle(private val moduleConfig: ModuleMetadata) {
             // sounds complicated.)
         }
 
-        val gradleDep = GradleDependencyInfo(TERASOLOGY_MODULES_GROUP, gestaltDependency.id.toString(), gestaltDependency.versionRange().toString())
+        val version = if (gestaltDependency.versionPredicate() is VersionRange) {
+            gestaltDependency.versionPredicate().toString()
+        } else {
+            // TODO: gradle-compatible version expressions for gestalt dependencies
+            //     https://github.com/MovingBlocks/gestalt/issues/114
+            gestaltDependency.minVersion.toString();
+        }
```

Comment:
I'm kind of confused how does this affect the logic? seems to output the results toString and discard?

Reply:
`if` can be used as an expression in kotlin. Like a more verbose `? :` ternary operator.

Reply:
oh I see, that is interesting.

Reply:
that's super nice, and something I'm totally missing in Java 🤓